pkg/plugins/autodiscovery/dockercompose/utils.go

```diff
@@ -0,0 +1,85 @@
+package dockercompose
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+	goyaml "gopkg.in/yaml.v3"
+)
+
+// searchDockerComposeFiles will look, recursively, for every files named Chart.yaml from a root directory.
+func searchDockerComposeFiles(rootDir string, filePatterns []string) ([]string, error) {
+
+	dockerComposeFiles := []string{}
+
+	// To do switch to WalkDir which is more efficient, introduced in 1.16
+	err := filepath.Walk(rootDir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			fmt.Printf("prevent panic by handling failure accessing a path %q: %v\n", path, err)
+			return err
+		}
+
+		for _, f := range filePatterns {
+			match, err := filepath.Match(f, info.Name())
+			if err != nil {
+				logrus.Errorln(err)
+				continue
+			}
+			if match {
+				dockerComposeFiles = append(dockerComposeFiles, path)
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Debugf("%d potential Docker compose file(s) found", len(dockerComposeFiles))
+
+	return dockerComposeFiles, nil
+}
```

Comment:
```suggestion
// searchDockerComposeFiles will look, recursively, for every files named Chart.yaml from a root directory.
func searchDockerComposeFiles(rootDir string, filePatterns []string) ([]string, error) {
	dockerComposeFiles := []string{}

	err := filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
		if err != nil {
			fmt.Printf("prevent panic by handling failure accessing a path %q: %v\n", path, err)
			return err
		}

		if !d.IsDir() {
			for _, f := range filePatterns {
				match, err := filepath.Match(f, d.Name())
				if err != nil {
					logrus.Errorln(err)
					continue
				}
				if match {
					dockerComposeFiles = append(dockerComposeFiles, path)
				}
			}
		}

		return nil
	})

	if err != nil {
		return nil, err
	}

	logrus.Debugf("%d potential Docker compose file(s) found", len(dockerComposeFiles))

	return dockerComposeFiles, nil
}
```

Implementation with `WalkDir` if more efficient (I went from 0.9s to 0.6s for the associated test)